app/src/poi.js

```diff
@@ -31,6 +31,42 @@ export default function Poi() {
         },
       });
 
+      const popup = new mapboxgl.Popup({
+        closeButton: false,
+        closeOnClick: false,
+      });
+      map.current.on("mousemove", "points-of-interest", (e) => {
+        // Change the cursor to a pointer when the mouse is move over the points-of-interest layer.
+        map.current.getCanvas().style.cursor = "pointer";
+        // Copy coordinates array.
+        const coordinates = e.features[0].geometry.coordinates.slice();
+        const name = e.features[0].properties.business_name;
+        const address =
+          e.features[0].properties.address +
+          ", " +
+          e.features[0].properties.city +
+          ", " +
+          e.features[0].properties.state +
+          ", " +
+          e.features[0].properties.zip +
+          ", " +
+          e.features[0].properties.country;
+        const category =
+          e.features[0].properties.category_name +
+          " (" +
+          e.features[0].properties.category_id +
+          ") ";
+        popup
+          .setLngLat(coordinates)
+          .setHTML("<p>" + name + "<br>" + address + "<br>" + category + "</p>")
+          .addTo(map.current);
```

Comment:
before adding it to map, should we check whether it already exists or not?

Reply:
https://github.com/mapbox/mapbox-gl-js/blob/16756b331d8d0d8aa63ef02bccdef77d748ba3a8/src/ui/marker.js#L189

 line number 189 of this js file. 
.addTo Don't add data in map again.

Reply:
great! then we don't need to do this.